synapse/storage/databases/main/devices.py

```diff
@@ -1245,8 +1245,8 @@ async def get_device_list_changes_in_rooms(
 
         def _get_device_list_changes_in_rooms_txn(
             txn: LoggingTransaction,
-            clause,
-            args,
+            clause: str,
+            args: List[Any],
```

Comment:
I think?

```suggestion
            args: List[str],
```

Reply:
It has both ints and strings in.
The rationale for using `Any` is that what should be in `args` depends on what's in `clause` so it doesn't make sense to be too picky.

Reply:
At quick glance it looked like just strings. 👍